lib/src/mamac_device_commands.dart

```diff
@@ -72,3 +77,79 @@ class EditDevice extends SimpleNode {
     (parent as MamacDeviceNode).update(params);
   }
 }
+
+class GetLogsNode extends SimpleNode {
+  static const String isType = 'getLogsNode';
+  static const String pathName = 'Get_Logs';
+  static const String childNodePrefix = 'Logs_';
+
+  static Map<String, dynamic> definition() => {
+        r'$is': isType,
+        r'$name': 'Get Logs',
+        r'$invokable': 'write',
+        r'$params': [],
+        r'$columns': [
+          {'name': 'success', 'type': 'bool', 'default': false},
+          {'name': 'message', 'type': 'string', 'default': ''}
+        ]
+      };
+
+  MamacDevice _device;
+  GetLogsService _getLogsService = new GetLogsService();
+  LinkProvider _link;
+
+  void set device(MamacDevice device) {
+    _device = device;
+
+    if (device.logPaths.isEmpty) {
+      return;
+    }
+
+    var entries = device.logPaths.map((LogEntry e) => e.displayName).join(',');
+
+    var paramsNode = [
+      {'name': 'logEntry', 'type': 'enum[${entries}]'}
+    ];
+
+    configs[r'$params'] = paramsNode;
+  }
+
+  GetLogsNode(String path, this._link) : super(path) {
```

Comment:
Since this command doesn't need to save the link configuration, you can remove the LinkProvider parameter from here and from the constructor call in profiles (in run.dart)